### PR TITLE
CWMS-1875 - Porting over cwbi-auth-http-client module from cumulus repo

### PR DESCRIPTION
**Pull request by rma-bryson**
Porting over cwbi-auth-http-client module from: https://github.com/HydrologicEngineeringCenter/cumulus-client/tree/main/cwbi-auth-http-client

History in the repository was not carried forward due to how limited the history was and level of effort to preserve history

### CODE REVIEW

File: cwbi-auth-http-client/build.gradle [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+plugins {
+    id "cwms-data-api-client.java-conventions"
+    id "cwms-data-api-client.deps-conventions"
+    id "cwms-data-api-client.publishing-conventions"
+}
+
+dependencies {
+    api(project(":cwms-http-client"))
+    api(project(":cwms-radar-model"))
```
Comment (adamkorynta) [inline cwbi-auth-http-client/build.gradle:33]: what does this pull from the data model? those should all be CDA data objects

File: cwbi-auth-http-client/build.gradle [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+publishing {
+    publications {
+        maven(MavenPublication) {
+            artifactId = "cwbi-auth-http-client"
+            from components.java
+        }
+    }
+}
+
+repositories {
```
Comment (adamkorynta) [inline cwbi-auth-http-client/build.gradle:55]: this already comes from `id "cwms-data-api-client.deps-conventions"`

**Review by adamkorynta — CHANGES_REQUESTED** [review]

File: cwbi-auth-http-client/build.gradle (hunk 1 above)
Comment (rma-bryson) [reply cwbi-auth-http-client/build.gradle]: yeah, that's not needed. Removed it

File: cwbi-auth-http-client/build.gradle (hunk 2 above)
Comment (rma-bryson) [reply cwbi-auth-http-client/build.gradle]: removed

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**